codec-redis/src/main/java/io/netty/handler/codec/redis/BulkStringRedisMessage.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.util.AbstractReferenceCounted;
+import io.netty.util.internal.StringUtil;
+
+/**
+ * Bulk Strings of <a href="http://redis.io/topics/protocol">RESP</a>
+ */
+public class BulkStringRedisMessage extends AbstractReferenceCounted implements RedisMessage {
+
+    public static final BulkStringRedisMessage NULL_BULK_STRING = new BulkStringRedisMessage(null);
+    public static final BulkStringRedisMessage EMPTY_BULK_STRING = new BulkStringRedisMessage(Unpooled.EMPTY_BUFFER);
+
+    private ByteBuf content;
+
+    public BulkStringRedisMessage(ByteBuf content) {
+        this.content = content;
+        if (content != null) {
```

Comment:
This guard requires the user to always check for `null` or the `refCount`, it's not possible to just check if the type is `instanceOf ReferenceCounted` and then call `release()` but it's always required to check if the content is `null`. How about always incrementing `refCount` of `BulkStringRedisMessage`?